core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -179,6 +179,21 @@ package object config {
   private[spark] val EVENT_LOG_CALLSITE_LONG_FORM =
     ConfigBuilder("spark.eventLog.longForm.enabled").booleanConf.createWithDefault(false)
 
+  private[spark] val EVENT_LOG_ENABLE_ROLLING =
+    ConfigBuilder("spark.eventLog.rolling.enabled")
+      .doc("Whether rolling over event log files is enabled.  If set to true, it cuts down " +
+        "each event log file to the configured size.")
+      .booleanConf
+      .createWithDefault(false)
+
+  private[spark] val EVENT_LOG_ROLLING_MAX_FILE_SIZE =
+    ConfigBuilder("spark.eventLog.rolling.maxFileSize")
+      .doc("The max size of event log file to be rolled over.")
```

Comment:
Sorry leaving a comment late like this but it should have been better to say this configuration is only effective when `spark.eventLog.rolling.enabled` is enabled.

Reply:
I think there're counter examples in Spark configurations which rely on the fact once there's a configuration `.enabled`, others are effective only when that is enabled.

Even we only check from the SHS configuration, `spark.history.fs.cleaner.*`, `spark.history.kerberos.*`, `spark.history.ui.acls.*` fall into the case.

Reply:
Hm, I tend to disagree with omitting such dependent configurations in their documentations. Can we add and link related configurations in the documentations?

Reply:
Sorry. Looks like we'll have to agree to disagree then. No one has privilege to make someone do the work under his/her authorship which he/she disagrees with - it will end up putting wrong authorship on commit.

Reply:
@HeartSaVioR, it had to be reviewed. I just happened to review and leave some comments late. Logically if that's not documented, how do users know what configuration is effective when? At least I had to read the codes to confirm.

Also, I am trying to make sure we're on the same page so I wouldn't happen to leave this comment again since you are a regular contributor. I don't think this is a good pattern to don't document the relationship between configurations. I am going to send an email to the dev list.

Reply:
@HyukjinKwon 
Thanks for initiating the thread in dev mailing list. I'm following up the thread and will be back once we get some sort of consensus.